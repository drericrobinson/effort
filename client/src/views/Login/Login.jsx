import React, { Component } from 'react';
import { Link } from '@reach/router';
import { AuthConsumer } from '../../utils/AuthContext';

export class Login extends Component {
  render() {
    return (
      <AuthConsumer>
        {({ isAuth, login, logout }) => (
          <div>
            <h3>
              <Link to='/'>Home</Link>
            </h3>
            {isAuth ? (
              <ul>
                <Link to='/landing'>Landing</Link>
                <button onClick={logout}>logout</button>
              </ul>
            ) : (
              <button onClick={login}>login</button>
            )}
          </div>
        )}
      </AuthConsumer>
    );
  }
}
