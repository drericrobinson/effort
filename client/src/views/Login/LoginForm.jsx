import React from 'react';
import { Formik } from 'formik';
import { UikFormInputGroup, UikInput, UikButton } from '../../@uik';
import { navigate } from '@reach/router';

export const LoginForm = props => {
  return (
    <Formik
      initialValues={{ email: '', password: '' }}
      validate={values => {
        let errors = {};
        if (!values.email) {
          errors.email = 'Required';
        } else if (
          !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
        ) {
          errors.email = 'Invalid email address';
        }
        return errors;
      }}
      onSubmit={values => {
        props.onLogin(values);
        navigate('/');
      }}
    >
      {({
        values,
        errors,
        touched,
        handleChange,
        handleBlur,
        handleSubmit,
        isSubmitting
      }) => (
        <form onSubmit={handleSubmit}>
          <UikFormInputGroup>
            <UikInput
              label='Email'
              type='email'
              name='email'
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.email}
            />
            {errors.email && touched.email && errors.email}
            <UikInput
              label='Password'
              type='password'
              name='password'
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.password}
            />
            {errors.password && touched.password && errors.password}
            <UikButton primary type='submit' disabled={isSubmitting}>
              Login
            </UikButton>
          </UikFormInputGroup>
        </form>
      )}
    </Formik>
  );
};
