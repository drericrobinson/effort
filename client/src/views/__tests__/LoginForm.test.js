import React from 'react';
import { render } from 'react-testing-library';
import 'jest-dom/extend-expect';
import { LoginForm } from '../Login/LoginForm';

describe('LoginForm', () => {
  it('renders the component', () => {
    const tree = render(<LoginForm />);
    expect(tree).toMatchSnapshot();
  });
});
