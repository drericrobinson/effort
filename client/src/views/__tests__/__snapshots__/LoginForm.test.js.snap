// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="uik-form-input-group__vertical"
        >
          <div
            class=""
          >
            <span
              class="uik-content-title__wrapper"
            >
              Email
            </span>
            <div
              class="uik-input__inputWrapper"
            >
              <input
                class="uik-input__input"
                name="email"
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            class=""
          >
            <span
              class="uik-content-title__wrapper"
            >
              Password
            </span>
            <div
              class="uik-input__inputWrapper"
            >
              <input
                class="uik-input__input"
                name="password"
                type="password"
                value=""
              />
            </div>
          </div>
          <button
            class="uik-btn__base uik-btn__primary"
            type="submit"
          >
            <span
              class="uik-btn__content"
            >
              Login
            </span>
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="uik-form-input-group__vertical"
      >
        <div
          class=""
        >
          <span
            class="uik-content-title__wrapper"
          >
            Email
          </span>
          <div
            class="uik-input__inputWrapper"
          >
            <input
              class="uik-input__input"
              name="email"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class=""
        >
          <span
            class="uik-content-title__wrapper"
          >
            Password
          </span>
          <div
            class="uik-input__inputWrapper"
          >
            <input
              class="uik-input__input"
              name="password"
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          class="uik-btn__base uik-btn__primary"
          type="submit"
        >
          <span
            class="uik-btn__content"
          >
            Login
          </span>
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
